src/engine/source/api/src/test/handlers.cpp

```diff
@@ -0,0 +1,1139 @@
+        const auto debugMode = eRequest.has_debug_mode() ? eRequest.debug_mode() : eTest::DebugMode::OUTPUT_ONLY;
+
+        const std::string defaultQueue {static_cast<char>(TEST_DEFAULT_PROTOCOL_QUEUE)};
+        const std::string strProtocolQueue {(eRequest.has_protocol_queue()) ? eRequest.protocol_queue() : defaultQueue};
+        if (eRequest.protocol_queue().size() > 1)
+        {
+            return ::api::adapter::genericError<ResponseType>("Protocol queue must be a single character long");
+        }
+
+        const uint8_t protocolQueue {static_cast<uint8_t>(strProtocolQueue.at(0))};
+        if (TEST_MAX_PROTOCOL_QUEUE < protocolQueue)
+        {
+            return ::api::adapter::genericError<ResponseType>(
+                fmt::format("Protocol queue ({}) must be a value between {} and {}",
+                            protocolQueue,
+                            TEST_MIN_PROTOCOL_QUEUE,
+                            TEST_MAX_PROTOCOL_QUEUE));
+        }
+
+        const auto protocolLocation =
+            eRequest.has_protocol_location() ? eRequest.protocol_location() : TEST_DEFAULT_PROTOCOL_LOCATION;
+
+        // Set debug mode
+        DebugMode routerDebugMode;
+        switch (debugMode)
+        {
+            case eTest::DebugMode::OUTPUT_AND_TRACES: routerDebugMode = DebugMode::OUTPUT_AND_TRACES; break;
+            case eTest::DebugMode::OUTPUT_AND_TRACES_WITH_DETAILS:
+                routerDebugMode = DebugMode::OUTPUT_AND_TRACES_WITH_DETAILS;
+                break;
+            case eTest::DebugMode::OUTPUT_ONLY:
+            default: routerDebugMode = DebugMode::OUTPUT_ONLY;
+        }
+
+        const auto assetTrace = eRequest.has_asset_trace() ? eRequest.asset_trace() : std::string();
+
+        // Get session
+        const auto session = sessionManager->getSession(eRequest.name());
+        if (!session.has_value())
+        {
+            return ::api::adapter::genericError<ResponseType>(fmt::format(SESSION_NOT_FOUND_MSG, eRequest.name()));
+        }
+
+        // Event in Wazuh format
+        const auto eventFormat =
+            fmt::format(WAZUH_EVENT_FORMAT, strProtocolQueue, protocolLocation, eRequest.event().string_value());
+        base::Event event;
+        try
+        {
+            event = base::parseEvent::parseWazuhEvent(eventFormat);
+        }
+        catch (const std::exception& e)
+        {
+            return ::api::adapter::genericError<ResponseType>(e.what());
+        }
+
+        // Add session ID to the event to filter it on the route
+        const auto sessionID = session.value().getSessionID();
+        const auto formattedPath = json::Json::formatJsonPath(TEST_FIELD_TO_CHECK_IN_FILTER);
+        event->setInt(static_cast<int>(sessionID), formattedPath);
+
+        auto dataSync = session->getDataSync();
+        auto expected {false};
+        if (callbackRunning.compare_exchange_strong(expected, true))
+        {
+            std::unique_lock<std::mutex> lock(queueMutex);
```

Comment:
This mutex blocks all requests and should not. The callback functions are associated with the session, not the global scope, there is no race condition or global data to protect.